python_modules/dagster/dagster/_core/definitions/events.py

```diff
@@ -459,17 +467,38 @@ def __new__(
             check.tuple_param(asset_key, "asset_key", of_type=str)
             asset_key = AssetKey(asset_key)
 
+        check.opt_mapping_param(tags, "tags", key_type=str, value_type=str)
+        if any([not tag.startswith(SYSTEM_TAG_PREFIX) for tag in tags or {}]):
+            check.failed(
+                "Users should not pass values into the tags argument for AssetMaterializations. "
```

Comment:
Should we even expose this as a top-level argument then?

Would we be better off exposing some sort of sequence of partition dimension keys then, that internally get stored as tags?

Reply:
I thought about doing that too, but upon deserializing a materialization I found that there needed to be a tags argument since we're storing tags within a materialization.

We could decide not to serialize the tags within an asset materialization altogether, in which case we'd need to serialize the partition key as some sort of sequence of partition dimension keys, so that we can determine the partition dimensions without reading the tags. I felt that that made the APIs more awkward though